types/types.override.pb.go

```diff
@@ -1,7 +1,30 @@
 package types
 
-// XXX_WellKnownType is a hack -- please don't make a pattern out of this!
-// This is used to trick certain toolsets to treat this field like a WKT StringValue.
-// See https://github.com/golang/protobuf/blob/70c277a8a150a8e069492e6600926300405c2884/jsonpb/jsonpb.go#L157
-// and https://github.com/golang/protobuf/blob/70c277a8a150a8e069492e6600926300405c2884/jsonpb/jsonpb.go#L191
-func (*UUIDValue) XXX_WellKnownType() string { return "StringValue" }
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/protobuf/jsonpb"
+)
+
+// MarshalJSONPB overloads UUIDValue's standard PB -> JSON conversion
+func (m *UUIDValue) MarshalJSONPB(*jsonpb.Marshaler) ([]byte, error) {
+	return []byte(fmt.Sprintf(`"%s"`, m.Value)), nil
+}
+
+// UnmarshalJSONPB overloads UUIDValue's standard JSON -> PB conversion
+func (m *UUIDValue) UnmarshalJSONPB(_ *jsonpb.Unmarshaler, data []byte) error {
+	m.Value = strings.Trim(string(data), `"`)
+	return nil
+}
+
+// MarshalJSONPB overloads JSONValue's standard PB -> JSON conversion
+func (m *JSONValue) MarshalJSONPB(*jsonpb.Marshaler) ([]byte, error) {
+	return []byte(m.Value), nil
```

Comment:
I know that you asked me about whether to validate this here or not and I said "not", but looking at it the way it is now, I'm thinking I was wrong.  In the big picture, what happens when these `Marshal`/`Unmarshal` functions return an error?

Reply:
I don't think we need to validate here, because the overall JSON is already validated for syntax before being parsed into the individual fields. I'm adding some negative tests that will demonstrate this.

Reply:
what validates it? the gRPC server?

Reply:
https://github.com/golang/protobuf/blob/3a3da3a4e26776cc22a79ef46d5d58477532dede/jsonpb/jsonpb.go#L653-L670 The Unmarshaler uses the standard golang encoding/json to decode the JSON before converting the raw JSON to Protobuf.